File: Labs/Lin_Lab4/die.py
"""
This module is a class that represents a die, that can be rolled and has a value.

Author: James Lin
Honor Code: I affirm that I have carried out the attached academic endeavors with full academic honesty,
in accordance with the Union College Honor Code and the course syllabus.
"""

import random


class Die:
    """
    Represents a die with a specified number of sides.
    __sides: An int that represents the number of sides on the die.
    __value: An int that represents the current value showing on the die.
    """

    def __init__(self, sides=6):
        """
        Constructs a Die object with a specified number of sides.
        Param: sides - The number of sides on the die. Defaults to 6.
        """
        self.__value = None
        self.__sides = sides
    
    def roll(self):
        """
        Rolls the die and sets the current value to the random number from 1 to the number of sides.
        """
        self.__value = random.randint(1, self.__sides)

    def get_value(self):
        """
        Returns the current value showing on the die.
        Returns: An int that represents the current value on the die.
        """
        return self.__value 

    def get_sides(self):
        """
        Returns the number of sides on the die.
        Returns: An int that represents the number of sides on the die.
        """
        return self.__sides
    

File: Labs/Lin_Lab7/src/Die.java
import java.util.Random;

/**
 * A die with a fixed number of sides and a current face value. The die is mutable.
 */
public class Die {
    private final int DEFUALT_SIDES = 6;
    private final int DEFUALT_VALUE = 1;


    private int value;
    private int sides;

    /**
     * Constructs a new Die object.
     * @param sides The int representation of sides in a Die.
     */
    public Die(int sides){

        this.value = DEFUALT_VALUE;
        this.sides = sides;
    }

    /**
     * Constructs a new die with the default number of sides (6).
     */
    public Die() {
        this.value = DEFUALT_VALUE;
        this.sides = DEFUALT_SIDES;
    }

    /**
     * Rolls this die, updating its value to a random integer in [1, sides].
     */
    public void roll(){
        Random r = new Random();
        this.value = r.nextInt(this.sides) + 1;
    }

    /**
     * Returns the value of the current Die
     * @return this.value The current value of the dice
     */
    public int getValue(){
        return this.value;
    }



//
//    /**
//     * @return this.sides The current side of the dice
//     */
//    public int getSides() {
//        return this.sides;
//    }
}
